go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -289,13 +295,43 @@ func (npw *nodePortWatcher) updateServiceFlowCache(service *kapi.Service, add bo
 
 				npw.ofm.updateFlowCacheEntry(key, nodeportFlows)
 			} else {
-				npw.ofm.updateFlowCacheEntry(key, []string{
-					fmt.Sprintf("cookie=%s, priority=110, in_port=%s, %s, %s=%s, tp_dst=%d, "+
-						"actions=%s",
-						cookie, npw.ofportPhys, flowProtocol, nwDst, externalIP, svcPort.Port, actions),
-					fmt.Sprintf("cookie=%s, priority=110, in_port=%s, %s, %s=%s, tp_src=%d, "+
-						"actions=output:%s",
-						cookie, npw.ofportPatch, flowProtocol, nwSrc, externalIP, svcPort.Port, npw.ofportPhys)})
+				// forward ARP requests to the node local IP (bypass OVN to avoid ARP reply / NA storm
+				// this will generate duplicate flows for each IP (one per key) but OFM and OVN will combine
```

Comment:
what do you mean by OFM and OVN will combine them into a single flow? you mean in the datapath?

Reply:
Yes, exactly.

The code adds an ARP / NS bypass flow for each tuple IP/protocol/portnumber. So if you create externalIP services as in my test case, with:
svc a, IP x.x.x.x, TCP 81, UDP 91  -> maps to ARP bypass for x.x.x.x
svc b, IP x.x.x.x TCP 82, UDP 92   -> maps to the same ARP bypass for x.x.x.x

Internally, this will add 4 keys in OFM via npw.ofm.updateFlowCacheEntry(key

So, there will be the exact same ARP / NS bypass 4 times, just under different keys under the map, and with different cookies. In OVS, if you add the exact same flow (with different cookies) multiple times, OVS will only apply the last flow and use the cookie of the last flow. So the flow will not be added 4 times. So that's an optimization that OVS (or the OpenFlow standard) does.
I was afraid that if I removed svc b, then the flow would be removed from OVS and the flow would be gone for svc a, as well. 
It turns out that the abstraction through OFM takes care of that, and there will still be the 2 duplicate ARP/NS bypass flows for svc a in the 2 other fields of the map, and the flow will still exist in OVS, just with a different cookie. I am not sure if this transaction does a replace or very briefly removes and readds the flow, but I'm probably worrying too much about that one.